test/models/user_test.rb

```diff
@@ -0,0 +1,27 @@
+require 'test_helper'
+
+class UserTest < ActiveSupport::TestCase
```

Comment:
@MagHz Test-driven development is an important development workflow concept.  It's great that you're already writing tests.  I'm not sure that you need it in this instance, however.  The things to test might be the interaction between the sign-in, sign-out and registration forms.  Here you're duplicating some of the tests that already exist in the Devise gem.